.github/workflows/ci.yaml

```
@@ -1088,25 +1088,18 @@ jobs:
           pattern: coverage-*
       - name: Upload coverage to Codecov (full coverage)
         if: needs.info.outputs.test_full_suite == 'true'
-        uses: Wandalen/wretry.action@v3.1.0
+        uses: codecov/codecov-action@v4.3.0
         with:
-          action: codecov/codecov-action@v4.3.0
-          with: |
-            fail_ci_if_error: true
-            flags: full-suite
-            token: ${{ env.CODECOV_TOKEN }}
-          attempt_limit: 5
-          attempt_delay: 30000
+          fail_ci_if_error: true
+          flags: full-suite
+          token: ${{ env.CODECOV_TOKEN }}
```

Not sure how it's set up on the GitHub/repo side, but mind changing all occurrences of `env` to `vars` for a test?
(assuming the Codecov token is set up as an environment variables -- IMO, we should use a secret with v4 though)
```suggestion
          token: ${{ vars.CODECOV_TOKEN }}
```